src/sagemaker/estimator.py

```diff
@@ -2219,7 +2219,20 @@ def _validate_and_set_debugger_configs(self):
         ):
             self.debugger_hook_config = DebuggerHookConfig(s3_output_path=self.output_path)
         elif not self.debugger_hook_config:
-            self.debugger_hook_config = None
+            self.debugger_hook_config = False
```

Comment:
Why are we changing this variable to `False`? Is this intentional? If so, why?

Reply:
The intent of the line above is to disable the debug hook config.
`False` is a more explicit indicator of that than `None

Reply:
`None` is already a falsy value in Python.

It doesn't make sense to change this to a boolean, since the `debugger_hook_config` variable can hold an actual config dict. You can simply check if `debugger_hook_config` is `None` instead of checking if it's `False`.

Reply:
[Documentation](https://sagemaker.readthedocs.io/en/stable/amazon_sagemaker_debugger.html#default-behavior-and-opting-out) however indicates to customers that they need to pass `False` to opt-out of debugger.

`None` is the values of this variable before we reach this code-block.

I'm making sure that when we check the value of post the `_validate` function call, we're indicative that we want debugger off instead of `None` which might come to mean that it has not yet been initialized.

Reply:
Yes, you're right: https://github.com/aws/sagemaker-python-sdk/blob/b66cb98b9223b437b02973197d62511776ccd9cb/src/sagemaker/tensorflow/estimator.py#L361-L365

Thank you for clarifying.